src/Markup/Avalonia.Markup/Data/BindingBase.cs

```diff
@@ -76,14 +76,14 @@ public BindingBase(BindingMode mode = BindingMode.Default)
 
         protected abstract ExpressionObserver CreateExpressionObserver(
             IAvaloniaObject target,
-            AvaloniaProperty targetProperty,
+            AvaloniaProperty? targetProperty,
             object? anchor,
             bool enableDataValidation);
 
         /// <inheritdoc/>
-        public InstancedBinding Initiate(
+        public InstancedBinding? Initiate(
             IAvaloniaObject target,
-            AvaloniaProperty targetProperty,
+            AvaloniaProperty? targetProperty,
```

Comment:
With what conditions we accept null property and null binding instance?

Reply:
Okay, it seems instanced binding can be expected "A InstancedBinding or null if the binding could not be resolved."

Reply:
Yeah the two are separate:

- `targetProperty` can be omitted if you can be sure you're not binding to `DataContext`
- `Initiate` can return `null` if it determines that the binding definitely can't be resolved on the target object